myia/frontends/pytorch_functions.py

```diff
@@ -35,12 +35,44 @@
 # HELPER FUNCTIONS ##########################################################
 
 
+@macro
+async def _denom(info, shp_ref, dtype_ref, dim_ref, unbiased_ref):
+    shp = build_value(await shp_ref.get())
+    dim = build_value(await dim_ref.get())
+    unbiased = build_value(await unbiased_ref.get())
+
+    def prod(_x):
+        return reduce(operator.mul, _x, 1)
+
+    if dim is None:
+        denom = int(prod(shp))
+    else:
+        denom = shp[dim]
+
+    if unbiased is True:
+        denom = denom - 1
+
+    return Constant(denom)
+
+
 @macro
 async def _dim_explicit(info, a_shp_ref, dim_ref):
     a_shp = build_value(await a_shp_ref.get())
     dim = build_value(await dim_ref.get())
-    if dim == -1:
-        dim = len(a_shp) - 1
+    if dim is None:
+        return Constant(dim)
+
+    if dim < 0:
+        dim = len(a_shp) + dim
+    return Constant(dim)
+
+
+@macro
+async def _dim_explicit_unsqueeze(info, a_shp_ref, dim_ref):
```

Comment:
Can you try to convert these macros to `@myia_static`? If you're going to use build_value on all arguments and return a constant, myia_static is terser and more robust.